### PR TITLE
Add analysis of eval usage for the whole data set, using Spark.

### PR DESCRIPTION
**Pull request by Tyler-R**
The analysis looks at the following aspects and compared them to what was found in the previous sample based analysis:
- How many function calls are created using eval.
- How many web pages use eval.
- How many scripts with function created using eval are hosted on different domains than the web page that uses them.

### CODE REVIEW

File: analyses/eval_analysis_spark.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+The previous Analysis can be found here: [eval_analysis.ipynb](eval_analysis.ipynb) and [eval_analysis.md](eval_analysis.md). An explanation for what eval is and a justification for why we are looking at eval usage can be found in the previous analysis.
+
+Similairly to in the other analysis, we will refer to using eval, rather than both eval and new Function, for simplicity.
+
+After looking at the entire dataset we found:
+1. 3.72% of function calls are created using eval.
+2. 8.76% of web pages have atleast 1 function call created using eval.  This is very similair to the samples taken in the previous analysis which was 8.92%.
+3. 32.65% of scripts hosting functions created using eval are hosted on a different domain that the domain of the web page that uses them. This is rounghly half of what was found in the earlier sample based analysis.
```
Comment (dzeber) [inline analyses/eval_analysis_spark.md:13]: Any thoughts as to why this is so different from the previous analysis? This seems too stark to be due to sampling variability.

File: analyses/eval_analysis_spark.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+
+### Eval and Function Analysis with Spark
+
+This is a continuation of the analysis on the usage of Eval and the Function class to create Javascript functions for web pages in the webcrawl dataset. This analysis uses Spark to look at the entire dataset, as apposed to the other analysis which looked at random samples in the dataset. The primary purpose of this analysis is to compare an analysis of the entire dataset with the previous analysis of small samples. This is being done to check if the small sample was representative of the entire dataset.  
+
+The previous Analysis can be found here: [eval_analysis.ipynb](eval_analysis.ipynb) and [eval_analysis.md](eval_analysis.md). An explanation for what eval is and a justification for why we are looking at eval usage can be found in the previous analysis.
```
Comment (dzeber) [inline analyses/eval_analysis_spark.md:6]: Nit: these links aren't working for me when viewing the notebook rendered on Github. I'd just use the full github.com URL to these files.

**Review by dzeber — APPROVED** [review]
Nice work! The introduction on eval usage in the previous notebook is great - I think it hits the right level of detail.

File: analyses/eval_analysis_spark.md (hunk 1 above)
Comment (Tyler-R) [reply analyses/eval_analysis_spark.md:13]: It looks like I made a mistake in the previous notebook.  When I was processing the functions  that were created using eval on each web page, I created a JSON object containing script URL's as keys and a count of the number of times that script URL was the source of a eval created function as the value.  When doing the analysis phase I accidentally only looked at the number of keys, not the function count.  

I corrected the issue and got around 33%, just like with spark. I will open up a new PR with the change in a bit.

Comment (Tyler-R) [reply analyses/eval_analysis_spark.md:13]: I opened up PR #10 which corrects the mistake.

**Pull request merged**